lcviewernoqt/drawitems/lcdimradial.cpp

```diff
@@ -34,21 +34,67 @@ void LCDimRadial::draw(LcPainter& painter, const LcDrawOptions &options, const l
     
     // Draw line
     EndCaps endCaps;
-    auto tLinePos = !mouseIsInside
-      ? mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), te.width + capSize)
-      : mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), -te.width - capSize);
-
-    auto tMText = !mouseIsInside
-      ? mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), te.width / 2 + capSize / 2)
-      : mousePos.move(mousePos - lc::geo::Coordinate(definitionPoint().x(), mousePos.y()), -te.width / 2 - capSize / 2);
-      
-    painter.move_to(definitionPoint2().x(), definitionPoint2().y());
-    painter.line_to(mousePos.x(), mousePos.y());
-    painter.line_to(tLinePos.x(), mousePos.y());
-    painter.stroke();
-    endCaps.render(painter, EndCaps::CLOSEDARROW, mousePos.x(), mousePos.y(), definitionPoint2().x(), definitionPoint2().y(), capSize) ;
-
-    this->drawText(value, textAngle(), lc::TextConst::AttachmentPoint::Top_center, tMText, painter, options, rect);
+
+    const bool textisInside = middleOfText().distanceTo(mousePos) < radiusCircle;
+    const bool textFitsInside = radiusCircle * 2. > te.width;
+
+    if (textisInside && textFitsInside) {
+        auto tLinePos = (definitionPoint2().x() > mousePos.x())
+                ? lc::geo::Coordinate(mousePos.x() - (te.width + capSize), mousePos.y())
+                : lc::geo::Coordinate(mousePos.x() + (te.width + capSize), mousePos.y());
+        auto tMText = (definitionPoint2().x() > mousePos.x())
+                ? lc::geo::Coordinate(mousePos.x() - (te.width/2 + capSize/2), mousePos.y() - (te.width/4 + capSize/2))
+                : lc::geo::Coordinate(mousePos.x() + (te.width/2 + capSize/2), mousePos.y() - (te.width/4 + capSize/2));
+
+        if (definitionPoint2().y() <= mousePos.y()) {
+            tMText = (definitionPoint2().x() > mousePos.x())
+                    ? lc::geo::Coordinate(mousePos.x() - (te.width/2 + capSize/2), mousePos.y())
+                    : lc::geo::Coordinate(mousePos.x() + (te.width/2 + capSize/2), mousePos.y());
+        }
+
+        painter.move_to(definitionPoint2().x(), definitionPoint2().y());
+        painter.line_to(mousePos.x(), mousePos.y());
+        painter.line_to(tLinePos.x(), mousePos.y());
+        painter.stroke();
+        endCaps.render(painter, EndCaps::CLOSEDARROW, mousePos.x(), mousePos.y(), definitionPoint2().x(), definitionPoint2().y(), capSize) ;
+
```

Comment:
Trying to cleanup a little bit here, so the code reads easier:

```C++
    using namespace lc::geo;

    Coordinate offsetX{te.width + capSize, 0};
    Coordinate offsetY{0, te.width/4 + capSize/2};

    if (textisInside && textFitsInside) {
	auto const offset = (definitionPoint2().x() > mousePos.x()) ? - offsetX : offsetX;
	auto const tLinePos = mousePos + offset;
	auto const tMText = mousePos + offsetX/2;

	if  (definitionPoint2().y() > mousePos.y()) {
		tMText -= offsetY;
	}

	//the painter code might read better with some adaptive interface
	auto move_to = [&painter] (Coordinate const& coord) {painter.move_to(coord.x, coord.y);}
	auto line_to = [&painter] (Coordinate const& coord) {painter.line_to(coord.x, coord.y);}

        move_to(definitionPoint2());
        line_to(mousePos);

```

the logic can be further simplified as far as the intention is still clear enough to be read from code.